sources/api/migration/migrator/tests/data/migration.rs

```diff
@@ -0,0 +1,47 @@
+use std::io::Write;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+/// This program serves as a pseudo-migration for testing. It will write its name and arguments to
+/// a file named `results.txt` one level above the directory it receives as the `--source-datastore`
+fn main() {
+    // the name of the program is lost with pentacle, so we hardcode a name here.
+    let mut message = "migration-name-replaceme:".to_owned();
+    let mut args = std::env::args();
+    let mut datastore_argument = String::new();
+    if let Some(_) = args.next() {
+        // i == 0 is the first arg after the binary path, i.e. shifted by one
+        for (i, arg) in args.enumerate() {
+            message.push(' ');
+            message.push_str(&arg);
+            if i == 2 {
+                // this is the position where we expect to find the datastore path, e.g.
+                // i ==>     0         1                  2        3                  4
+                // migration --forward --source-datastore /some/ds --target-datastore /another/ds
+                datastore_argument = arg.to_string();
+            }
+        }
+    }
```

Comment:
```suggestion
    let args = args.next().expect("Test migration missing 0th argument..?!")
    datastore_argument = args.nth(2).expect("Test migration not passed enough arguments, got {}", args.join(" "));
    message.push_str(args.join(" "));
```
Untested, but I feel like it can be simplified a bit.

Reply:
Didn't work on the first try, but...

> webern force-pushed the webern:pr/signed-migrations branch from 1f69203 to afc4b5d 13 seconds ago

I removed the large end-to-end migrator 'unit' test. I will try to introduce it in a future PR where we can spend more time refining it.